src/components/UserProfile.js

```diff
@@ -0,0 +1,67 @@
+import React, { useState } from 'react';
+import { Button, Card, CardBody, CardFooter, Table } from 'reactstrap';
+import { useNavigate } from 'react-router-dom'; // Import useNavigate hook
+
+const UserProfile = ({ user, updateProfileClick }) => {
```

Comment:
You pass in an `updateProfileClick` prop, but it's not actually used in this component?